packages/ui/src/components/ActivityToast/ActivityToast.tsx

```diff
@@ -0,0 +1,152 @@
+import { FC, useEffect, useMemo, useState } from "react"
+import { CloseIcon } from "../../icons/CloseIcon"
+
+type vertical = "top" | "center" | "bottom"
+type horizontal = "left" | "center" | "right"
+
+interface ActivityToastProps {
+  action: string
+  description: string
+  showToast?: boolean
+  closeToast: () => void
+  duration?: 1000 | 2000 | 3000 | 4000 | 5000
+  vertical?: vertical
+  horizontal?: horizontal
+}
+
+const durationMap = {
+  1000: "duration-1000",
+  2000: "duration-2000",
+  3000: "duration-3000",
+  4000: "duration-4000",
+  5000: "duration-5000",
+}
+
+const ActivityToast: FC<ActivityToastProps> = ({
+  action,
+  description,
+  showToast,
+  closeToast,
+  duration = 5000,
+  vertical = "bottom",
+  horizontal = "center",
+}) => {
+  return showToast ? (
+    <ActivityToastComponent
+      action={action}
+      description={description}
+      closeToast={closeToast}
+      duration={duration}
+      vertical={vertical}
+      horizontal={horizontal}
+    />
+  ) : null
+}
+
+const ActivityToastComponent: FC<ActivityToastProps> = ({
+  action,
+  description,
+  closeToast,
+  duration = 5000,
+  vertical = "bottom",
+  horizontal = "center",
+}) => {
+  const [isActive, setIsActive] = useState(true)
+  useEffect(() => {
+    // ensure that the toast is visible, setting the width to full for the progression bar
+    setTimeout(() => setIsActive(false), 10)
+    const timeout = setTimeout(() => closeToast(), duration)
+    return () => {
+      closeToast()
+      clearTimeout(timeout)
+    }
+  }, [])
+
+  const position = useMemo(() => {
+    let v, h
+
+    switch (horizontal) {
+      case "left":
+        h = `left-2`
+        break
+      case "center":
+        h = `left-2/4 transform -translate-x-2/4`
+        break
+      case "right":
+        h = `right-2`
+        break
+      default:
+        h = `left-2/4`
+        break
+    }
+
+    switch (vertical) {
+      case "top":
+        v = `top-2`
+        break
+      case "center":
+        v = `top-2/4 transform -translate-y-2/4`
+        break
+      case "bottom":
+        v = `bottom-2`
+        break
+      default:
+        v = `top-2/4`
+        break
+    }
+
+    return `${v} ${h}`
+  }, [vertical, horizontal])
+
+  return (
+    <div className={`fixed ${position}`}>
+      <div className="relative p-5 bg-white rounded-lg overflow-hidden shadow-lg border border-solid border-neutrals.200 w-96">
+        <div className="flex flex-1">
+          <div className="flex flex-col justify-center mr-4">
+            {/* TODO: remove hardcode - decide how to manage icons in notifications */}
```

Comment:
@vladutjs @ismaeldm95 should we let the dapp to decide the icon associated to the action or we choose from a preset (handing the logic for the action)

in the second case, the action would be an `enum` like
- SWAP
- TRANSFER
- MINT